### PR TITLE
[opt](function) Optimize the trim function for single-char inputs

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

before
```
mysql [test]>select count(ltrim(str,"1")) from stringDb2;
+------------------------+
| count(ltrim(str, '1')) |
+------------------------+
|               64000000 |
+------------------------+
1 row in set (7.79 sec)
```

now
```
mysql [test]>select count(ltrim(str,"1")) from stringDb2;
+------------------------+
| count(ltrim(str, '1')) |
+------------------------+
|               64000000 |
+------------------------+
1 row in set (0.73 sec)
```

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Review by HappenLee — DISMISSED** [review]
LGTM

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

File: be/src/util/simd/vstring_function.h [hunk 1]
```diff
@@ -100,169 +100,83 @@ class VStringFunctions {
...
-                int offset = __builtin_clz(~(mask << REGISTER_SIZE));
-                /// means not found
-                if (offset == 0) {
-                    return StringRef(str.data + begin, end - begin + 1);
-                } else {
-                    end -= offset;
+        const auto* p = end;
+#if defined(__AVX2__) || defined(__SSE4_2__)
+        if constexpr (trim_single) {
+            constexpr auto AVX2_BYTES = sizeof(__m256i);
```
Comment (zclllyybb) [inline be/src/util/simd/vstring_function.h:113]: use `AVX2_BYTES` for `__m256i` may be misleading. should choose another word.

File: be/src/util/simd/vstring_function.h [hunk 2]
```diff
@@ -100,169 +100,83 @@ class VStringFunctions {
...
+            const auto ch = remove_str.data[0];
+            const auto size = end - begin;
+            const auto* const avx2_begin = end - size / AVX2_BYTES * AVX2_BYTES;
+            const auto spaces = _mm256_set1_epi8(ch);
+            for (p = end - AVX2_BYTES; p >= avx2_begin; p -= AVX2_BYTES) {
+                uint32_t masks = _mm256_movemask_epi8(
+                        _mm256_cmpeq_epi8(_mm256_loadu_si256((__m256i*)p), spaces));
+                if ((~masks)) {
```
Comment (zclllyybb) [inline be/src/util/simd/vstring_function.h:121]: please add comment for `masks` and relative operations

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by wangbo — APPROVED** [review]
LGTM

**Pull request merged**